pkg/querier/batch/batch.go

```diff
@@ -85,51 +86,74 @@ func newIteratorAdapter(underlying iterator) chunkenc.Iterator {
 }
 
 // Seek implements chunkenc.Iterator.
-func (a *iteratorAdapter) Seek(t int64) bool {
+func (a *iteratorAdapter) Seek(t int64) chunkenc.ValueType {
 
 	// Optimisation: fulfill the seek using current batch if possible.
 	if a.curr.Length > 0 && a.curr.Index < a.curr.Length {
 		if t <= a.curr.Timestamps[a.curr.Index] {
 			//In this case, the interface's requirement is met, so state of this
 			//iterator does not need any change.
-			return true
+			return chunkenc.ValFloat
 		} else if t <= a.curr.Timestamps[a.curr.Length-1] {
 			//In this case, some timestamp between current sample and end of batch can fulfill
 			//the seek. Let's find it.
 			for a.curr.Index < a.curr.Length && t > a.curr.Timestamps[a.curr.Index] {
 				a.curr.Index++
 			}
-			return true
+			return chunkenc.ValFloat
 		}
 	}
 
 	a.curr.Length = -1
 	a.batchSize = 1
-	if a.underlying.Seek(t, a.batchSize) {
+	if typ := a.underlying.Seek(t, a.batchSize); typ != chunkenc.ValNone {
 		a.curr = a.underlying.Batch()
-		return a.curr.Index < a.curr.Length
+		if a.curr.Index < a.curr.Length {
+			return typ
+		}
 	}
-	return false
+	return chunkenc.ValNone
 }
 
 // Next implements chunkenc.Iterator.
-func (a *iteratorAdapter) Next() bool {
+func (a *iteratorAdapter) Next() chunkenc.ValueType {
 	a.curr.Index++
-	for a.curr.Index >= a.curr.Length && a.underlying.Next(a.batchSize) {
+	for a.curr.Index >= a.curr.Length && a.underlying.Next(a.batchSize) == chunkenc.ValFloat {
 		a.curr = a.underlying.Batch()
 		a.batchSize = a.batchSize * 2
 		if a.batchSize > chunk.BatchSize {
 			a.batchSize = chunk.BatchSize
 		}
 	}
-	return a.curr.Index < a.curr.Length
+
+	if a.curr.Index < a.curr.Length {
+		return chunkenc.ValFloat
+	}
+	return chunkenc.ValNone
 }
 
 // At implements chunkenc.Iterator.
 func (a *iteratorAdapter) At() (int64, float64) {
 	return a.curr.Timestamps[a.curr.Index], a.curr.Values[a.curr.Index]
 }
 
+// AtHistogram implements chunkenc.Iterator. Histogram support isn't complete yet,
+// so this function just returns (0, nil).
+func (a *iteratorAdapter) AtHistogram() (int64, *histogram.Histogram) {
+	return 0, nil
+}
+
+// AtFloatHistogram implements chunkenc.Iterator. Histogram suppport isn't complete
+// yet, so this function just returns (0, nil).
+func (a *iteratorAdapter) AtFloatHistogram() (int64, *histogram.FloatHistogram) {
+	return 0, nil
+}
```

Comment:
Until it is implemented, let's return `errors.New("not implemented")`, this will make sure later that we have not missed something.

Reply:
Returning error was a wrong suggestion. You can panic here for now like you did in some other place.